### ISSUE TITLE
Would it be possible to get rid of the dependency on GNU coreutils and GNU getopt on macOS?

### ISSUE THREAD

**Issue by s3rvac** [avast-tl/retdec#263]
Currently, we require macOS users to have [GNU coreutils](https://www.gnu.org/software/coreutils) and [GNU getopt](https://www.gnu.org/software/libc/manual/html_node/Getopt.html) in `PATH`, with the additional requirement that GNU getopt has to be in `PATH` before the built-in `getopt`. I wonder if it would be possible to get rid of these two dependencies and just use the built-in macOS versions of various commands.

Currently, I am aware of the following commands and places where the GNU versions are needed:
* `greadlink` in all our scripts inside `scripts/` to find the path to the directory where each script is located
* `getopt` in multiple scripts inside `scripts/` (two `retdec-*` scripts and scripts in `scripts/type_extractor`)
* `gtimeout` in `scripts/retdec-archive-decompiler.sh`
* `gsha256sum` in `cmake/install-share` (the use of `gsha256sum` in there is already optional if you have `shasum` in your `PATH`)

**Comment by psifertex** [comment]
`shasum` is included by default in the standard MacOS install, isn't it? That sound solve the last one.

**Comment by psifertex** [comment]
"should solve", sorry.

**Comment by s3rvac** [comment]
Yes, the last one (`sha256sum`) should have already been solved since 0f1281e536bfe2e6053f702138114de51066335d:
```sh
sha256hash()
{
    # To compute the SHA-256 hash, try several alternatives, based on what
    # command is available on the system.
    if command -v gsha256sum >/dev/null 2>&1; then
        gsha256sum "$@"
    elif command -v shasum >/dev/null 2>&1; then
        shasum -a 256 "$@"
    else
        sha256sum "$@"
    fi
}
```

**Comment by PeterMatula** [comment]
Fixed by replacing Bash scripts with Python (#147, #338).
